Renamed auditd module fields

andrewkroh opened the pull request:
Updates to the auditd module

- Update to latest go-libaudit.
- Fixed an issue where the proctitle value was being truncated.
- Fixed an issue where process args were incorrectly interpretted as hex data.
- Fixed parsing of the `key` value when multiple keys are present.
- Added a warning to the examples rules about the cost of auditing all network traffic.
- Changed the event format to allow for some common fields between modules.
- Updated fields.yml.

The dashboards need to be updated due to the field changes.

Closes #5423

ruflin commented on auditbeat/module/auditd/_meta/accept.json:
```diff
@@ -0,0 +1,76 @@
+        "action": "accepted-connection-from",
+        "category": "audit-rule",
+        "module": "auditd",
+        "type": "syscall"
```
Nice usage of `type`, module and category.

ruflin commented on auditbeat/module/auditd/_meta/accept.json:
```diff
@@ -0,0 +1,76 @@
+    "user": {
+        "auid": "unset",
+        "egid": "0",
+        "euid": "0",
```
These are the the ids which would also end up in `user.id: [...]` I assume?

andrewkroh replied:
Yes. Do you think I should add that now? I could add it later.

ruflin replied:
No need to add it now as it's just an addition / feature.

andrewkroh commented:
I updated this with fields.yml changes, squashed, and give it a useful commit message.

Please use the Rebase and Merge to keep the vendor changes separate from code changes.

ruflin commented on auditbeat/_meta/fields.common.yml:
```diff
@@ -3,7 +3,121 @@
+    description: >
+      The name of the module that generated the event.
+
+  - name: event.action
```
@MikePaquette That is an interesting field. I wonder if we can reuse this.

ruflin commented on auditbeat/_meta/fields.common.yml:
```diff
@@ -3,7 +3,121 @@
+      type: text
+      description: The path to the file.
+      multi_fields:
+      - name: raw
```
@andrewkroh Potentially something we should support in ECS by default.

ruflin commented on auditbeat/module/auditd/_meta/fields.yml:
```diff
@@ -2,51 +2,158 @@
+      description: The SELinux identity of the actor.
       fields:
-      - name: action
+      - name: user
```
I kind strange naming as it's `user.selinux.user`. Should this be account?

ruflin commented on auditbeat/module/auditd/_meta/fields.yml:
```diff
@@ -2,51 +2,158 @@
+      type: keyword
+      description: This is the path associated with a unix socket.
+
+  - name: network.direction
```
Fingers cross that this is not changing ;-)

andrewkroh replied:
I think we need to standardize on the enum values.

ruflin replied:
+1 on having a recommend standard.

ruflin commented on auditbeat/module/auditd/_meta/fields.yml:
```diff
@@ -176,35 +179,59 @@
           or command that was being executed that triggered the event.
-      - name: key
+
+    - name: paths
```
Should this be singular?

andrewkroh replied:
It is usually a list of raw path objects so I named with a plural. Is that the correct way to do it?

ruflin replied:
My argument is normally to make it singular as long as it's not an array, as the user that is accessing it only looks for a single item and the "key" for this entry reads correct. Here for example `paths.inode`, inode is only one path. But I have this conversation often with @exekias and we probably agreed to disagree 😆

ruflin commented on auditbeat/module/auditd/_meta/fields.yml:
```diff
@@ -821,6 +824,25 @@
+          - name: saddr
+            type: keyword
+            description: The raw socket address structure.
+          - name: addr
```
I suggest not to use abbreviation for address.

andrewkroh replied:
I opted for `addr` because that's the exact name coming from the kernel. But I can rename this is `go-libaudit`.

ruflin commented:
@andrewkroh This PR needs a make update.

andrewkroh commented:
I rebased it to fix some merge conflicts caused by me merging another PR.

ruflin commented:
@andrewkroh I merged this to get the major change in. We can still continue to discuss the minor details.
